spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -142,6 +147,24 @@ public Subscriber subscribe(String subscription, MessageReceiver messageReceiver
 				returnImmediately));
 	}
 
+	@Override
+	public <T> List<Message<T>> pullAndConvert(String subscription, Integer maxMessages, Boolean returnImmediately,
```

Comment:
@artembilan @dhoard I added this. Let me know how this feels.

Reply:
I have code in production that would benefit from this method ...

pseudo code ...
```
@Scheduled(cron = "0 * * * *")
public void runHourly() {
    List<JSONObject> list = pubSubSubscriber.pullAndConvert(...);

    // Convert to batch format
    // Send batch format to third party via HTTP

    // ack list of messages
}
``` 

This can be done with Spring Integration ... just requires more code.